clients/src/main/java/org/apache/kafka/common/protocol/MessageUtil.java

```diff
@@ -37,6 +38,14 @@ public static short serializedUtf8Length(CharSequence input) {
         return (short) count;
     }
 
+    public static byte[] byteBufferToArray(ByteBuffer buf) {
+        byte[] arr = new byte[buf.remaining()];
+        int prevPosition = buf.position();
+        buf.get(arr);
+        buf.position(prevPosition);
```

Comment:
nit: maybe use `finally` to restore this value.

Reply:
ok